File: cp_rust/src/bin/abc126-b-yymm-or-mmyy.rs
// Created by Ayush Biswas at 2026/06/09 19:41
// https://atcoder.jp/contests/abc126/tasks/abc126_b
use cp_lib::*;

// @code begin
use cpio::*;

fn is_month(part: &str) -> bool {
    let code: usize = part.parse().unwrap();
    code > 0 && code <= 12
}

sol! {
    fn solution(
        s: String
    ) -> String {
        let front_month = is_month(&s[0..=1]);
        let back_month = is_month(&s[2..=3]);
        if front_month && back_month {
            "AMBIGUOUS"
        } else if front_month {
            "MMYY"
        } else if back_month {
            "YYMM"
        } else {
            "NA"
        }.into()
    }
}

// @code end

File: cp_ocaml/src/bin/abc126_b_yymm_or_mmyy.ml
(* Created by Ayush Biswas at 2026/06/23 11:51
https://atcoder.jp/contests/abc126/tasks/abc126_b *)
open Core

let valid_month m = m >= 1 && m <= 12

let sol s =
  let a = Int.of_string @@ String.prefix s 2
  and b = Int.of_string @@ String.suffix s 2 in
  let res =
    match valid_month a, valid_month b with
    | true, true -> "AMBIGUOUS"
    | true, false -> "MMYY"
    | false, true -> "YYMM"
    | false, false -> "NA"
  in
  printf "%s" res
;;

let () = Scanf.scanf "%s" sol
